src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnectionManager.cs

```diff
@@ -53,5 +52,7 @@ public Task WriteAckableMessageAsync(ServiceMessage seviceMessage, CancellationT
 
             return _serviceConnection.WriteAckableMessageAsync(seviceMessage, cancellationToken);
         }
+
+        internal IServiceConnectionContainer GetServiceConnectionContainer() => _serviceConnection;  //for test only
```

Comment:
Is it possible to inherit this class in your test code? It's better not to create a function only for test.

Reply:
No... The `ServiceConnectionManager` is injected inside the method `CreateHubContextAsync`, so we can't replace it with a mocked one.